tmt/steps/__init__.py

```diff
@@ -62,6 +74,85 @@
 TEST_TOPOLOGY_FILENAME_BASE = 'tmt-test-topology'
 
 
+PHASE_OPTIONS = tmt.options.create_options_decorator([
+    option(
+        '--insert',
+        is_flag=True,
+        default=False,
+        help='Add this phase instead of overwriting the existing ones.'),
+    option(
+        '--update',
+        is_flag=True,
+        default=False,
+        help='Update existing phase. Use --name to specify which one.'),
+    option(
+        '--name',
+        type=str,
+        help="Name of the existing phase which should be updated when '--update' is used."),
```

Comment:
I'm afraid I don't follow, can you provide an example of the situation and what you would like to see happen?

Reply:
I have a plan

```
❯ cat plans/example.fmf
summary: Basic smoke test
discover:
    how: fmf
execute:
    how: tmt
report:
  - name: html
    how: html
  - how: display
```

I want to update the second report phase with `how: display`.
It is hard to find out the `name` attribute of it.
It is not displayed anywhere, the only place I was able to spot it was when running with `-ddd`:

```
❯ tmt plan show -ddd |& grep default-0
    DiscoverPlugin.delegate(step=discover, data=None, raw_data={'how': 'fmf', 'name': 'default-0'})
    DiscoverPlugin.delegate(step=discover, data=DiscoverFmfStepData(name='default-0', how='fmf', order=50, summary=None, where=[], dist_git_source=False, dist_git_type=None, url=None, ref=None, path=None, test=[], link=[], filter=[], exclude=[], modified_only=False, modified_url=None, modified_ref=None, dist_git_init=False, dist_git_remove_fmf_root=False, dist_git_merge=False, dist_git_extract=None, sync_repo=False, fmf_id=False, repository=None, revision=None), raw_data=None)
    ProvisionPlugin.delegate(step=provision, data=None, raw_data={'name': 'default-0', 'how': 'virtual'})
    ProvisionPlugin.delegate(step=provision, data=ProvisionTestcloudData(name='default-0', how='virtual', order=50, summary=None, hardware=None, role=None, guest=None, facts=GuestFacts(in_sync=False, arch=None, distro=None, kernel_release=None, package_manager=None, has_selinux=None, is_superuser=None, os_release_content={}, lsb_release_content={}), port=None, user='root', key=[], password=None, ssh_option=[], image='fedora', memory=2048, disk=40, connection='session', arch='x86_64', list_local_images=False, image_url=None, instance_name=None), raw_data=None)
    PreparePlugin.delegate(step=prepare, data=None, raw_data={'name': 'default-0', 'how': 'shell'})
    PreparePlugin.delegate(step=prepare, data=PrepareShellData(name='default-0', how='shell', order=50, summary=None, where=[], script=[]), raw_data=None)
    ExecutePlugin.delegate(step=execute, data=None, raw_data={'how': 'tmt', 'name': 'default-0'})
    ExecutePlugin.delegate(step=execute, data=ExecuteInternalData(name='default-0', how='tmt', order=50, summary=None, where=[], duration='1h', exit_first=False, script=[], interactive=False, no_progress_bar=False), raw_data=None)
    ReportPlugin.delegate(step=report, data=None, raw_data={'how': 'display', 'name': 'default-0'})
    ReportPlugin.delegate(step=report, data=ReportDisplayData(name='default-0', how='display', order=50, summary=None, display_guest='auto'), raw_data=None)
    FinishPlugin.delegate(step=finish, data=None, raw_data={'name': 'default-0', 'how': 'shell'})
    FinishPlugin.delegate(step=finish, data=FinishShellData(name='default-0', how='shell', order=50, summary=None, where=[], script=[]), raw_data=None)
```

While I guess this problem was not introduced in this MR, it is making it more visible, as I want to use the `--name` attribute to update a specific phase.

I would like to propose, in case of multiple phases for the step, display the name always, including the default one.
That would make this easy.

Reply:
Ah, now I understand :) Well, that's on purpose, actually, tmt will actively *hide* names looking like a default name. That's why `tmt plan show` does not print it out. It was discussed on IRC - I should be able to find the discussion if needed -, and it was a feature @psss wanted to keep around: https://github.com/teemtee/tmt/pull/1384#discussion_r945656170.

Showing the `name` key always, i.e. not "hiding" info from the user as I described it a year ago, would fit my nature. But, since @psss felts strongly about this, I'd not block this PR on changing this particular behavior, I'd rather do so in a separate patch to give @psss a chance to voice his concerns, although I believe the new `--update` requiring the knowledge of `name` changes the situation in favor of "always visible" behavior.

Reply:
+1, thanks for looking up the details. Let's not block on this.